web/modules/custom/custom_weather/src/Plugin/Block/CustomWeather.php

```diff
@@ -0,0 +1,33 @@
+<?php
+namespace Drupal\custom_weather\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+
+#[Block(
+  id: "custom_weather_block",
+  admin_label: new TranslatableMarkup("Custom Weather block"),
+)]
+
+class CustomWeather extends BlockBase {
+  public function build():array {
+    $weatherData = $this->getWeatherApi();
+    $temp = $weatherData['main']['temp'];
+    $weather_text = $weatherData['weather'][0]['main'];
+    $temp = round($temp - 273);
+
+    return [
+      '#theme' => 'custom_weather_block',
+      '#temp' => $temp,
+      '#weather_text' => $weather_text,
+    ];
+  }
+
+  protected function getWeatherApi():mixed {
```

Comment:
Logic shouldn't be in the Block plugin.
Move to the service.
Also, need to write tests for new service.